source/common/filter/config_discovery_impl.h

```diff
@@ -126,11 +117,11 @@ class DynamicFilterConfigProviderImpl : public DynamicFilterConfigProviderImplBa
     }
   }
 
+  const std::string& getStatPrefix() const { return stat_prefix_; }
+  Server::Configuration::FactoryContext& getFactoryContext() const { return factory_context_; }
```

Comment:
If these fields are only needed by the derived classes, then they should be protected (without accessors)

Reply:
Thanks for the comments. Done!